headers/xdp/xsk.h

```diff
@@ -201,8 +201,35 @@ struct xsk_umem_config {
 	__u32 frame_size;
 	__u32 frame_headroom;
 	__u32 flags;
+	__u32 tx_metadata_len;
 };
 
+/* The following fields are required, and must be set at once:
+ * 
+ * @umem_area, @fill, @comp
```

Comment:
Better to have the mandatory fields be direct function arguments, and have the options only be the, well, optional values. That way, a caller can also pass a NULL opts argument if they are happy with the defaults...